### PR TITLE
xeno weed changes

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
lowered the grow time limit from 15-20 seconds to 5-10 seconds
weeds can't grow in water now
documented the variables and procs
use the cooldown system
children weeds of the parent node will be destroyed on parent destroyed

weeds can still grow (just showing)
https://gyazo.com/e0e6b4ab35e0d7abd74981a7ddc976b2

weeds will die when their parent dies
https://gyazo.com/691f64e118a467b534fb1c5137684380

weeds finding a new parent
https://gyazo.com/14c17b365803a29e85f861c5d226cbf3

## Why It's Good For The Game
it feels awful clearing the xeno weeds at times, this should help that.
xeno weeds also feel really slow when growing, so this'll help xenos there.

## Changelog

:cl:
expansion: xeno weeds will now be destroyed when their parent node is destroyed
balance: xeno weeds spread cooldown lowered from 15-20 seconds to 5-10 seconds
/:cl:

### CODE REVIEW

File: code/game/objects/structures/aliens.dm [hunk 1]
```diff
@@ -144,30 +144,33 @@
 	smoothing_flags = SMOOTH_BITMASK
 	smoothing_groups = list(SMOOTH_GROUP_ALIEN_RESIN, SMOOTH_GROUP_ALIEN_WEEDS)
 	canSmoothWith = list(SMOOTH_GROUP_ALIEN_WEEDS, SMOOTH_GROUP_WALLS)
-	var/last_expand = 0 //last world.time this weed expanded
-	var/growth_cooldown_low = 150
-	var/growth_cooldown_high = 200
-	var/static/list/blacklisted_turfs
-
+	///the parent node that will determine if we grow or die
+	var/obj/structure/alien/weeds/node/parent_node
```
Comment (Fikou) [inline code/game/objects/structures/aliens.dm:148]: if there are multiple weed nodes near each other, the weeds will all be destroyed even though there is a node in range

Comment (jjpark-kb) [reply code/game/objects/structures/aliens.dm:148]: only the child weeds, which is intentional.

**Comment by Dorsisdwarf** [comment]
Feel like the parent weed needs a hp buff so it isnt 1 welder click to wipe several minutes of weed growth?

**Comment by ATH1909** [comment]
the weed receding looks way too fast to me

**Comment by ATH1909** [comment]
and it'd be nice if weeds wouldn't decay if there's another weed node nearby that could sustain them

like, if you plant a weed node, let it grow some weeds, plant a second weed node right next to it, and then destroy the first weed node, the weeds shouldn't all suddenly disintegrate

**Comment by jjpark-kb** [comment]
> Feel like the parent weed needs a hp buff so it isnt 1 welder click to wipe several minutes of weed growth?

I mean, maybe? I am not too sure if doing that is appropriate for this PR, but perhaps it is.

> the weed receding looks way too fast to me

I can increase the timer.

> and it'd be nice if weeds wouldn't decay if there's another weed node nearby that could sustain them
like, if you plant a weed node, let it grow some weeds, plant a second weed node right next to it, and then destroy the first weed node, the weeds shouldn't all suddenly disintegrate

I can do that.

**Pull request merged**